### PR TITLE
chore: Enhance PR labeling based on PR title

### PR DESCRIPTION
**Pull request by ermshiperete**
This PR adds the labeler-action GH action that applies labels based on the PR title or description. This allows to automatically assign labels based on our conventional commit message.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -85,3 +89,41 @@ windows/config/: windows/src/desktop/**
...
+    - '^.*\(common(/|\)):'
+  'developer/':
+    - '^.*\(developer\):'
+  'ios/':
+    - '^.*\(ios\):'
+  'linux/':
+    - '^.*\(linux\):'
+  'mac/':
+    - '^.*\(mac\):'
+  'web/':
```
Comment (darcywong00) [inline .github/labeler.yml:126]: What about oem from `resources/scopes/scopes.json`?

Comment (ermshiperete) [reply .github/labeler.yml:128]: Good point. Done.

**Review by darcywong00 — APPROVED** [review]
lgtm

File: .github/labeler.yml [hunk 2]
```diff
@@ -85,3 +89,43 @@ windows/config/: windows/src/desktop/**
...
+  'linux/':
+    - '\(.*linux.*\):'
+  'mac/':
+    - '\(.*mac.*\):'
+  'oem/':
+    - '\(.*oem.*\):'
+  'web/':
+    - '\(.*web.*\):'
+  'windows/':
+    - '\(.*windows.*\):'
```
Comment (mcdurdin) [inline .github/labeler.yml:131]: Generally the code changes check will find these scopes for us so this is probably not necessary.

File: .github/labeler.yml [hunk 3]
```diff
@@ -1,8 +1,12 @@
 #
-# This file adds labels based on the scopes in resources/scopes/scopes.json.
+# This file adds labels based on the scopes in resources/scopes/scopes.json and
+# resources/scopes/commit-types.json
 # Currently it must be manually kept up to date. Not all labels are added, only
 # common ones. The others are commented out. There is still some variance between
 # folder names and labels; consider this documentation of that ;-)
+
+#
+# The top half of the file adds labels based on changed files
```
Comment (mcdurdin) [inline .github/labeler.yml:9]:
```suggestion
# Add labels based on changed files using actions/labeler
```

File: .github/labeler.yml [hunk 4]
```diff
@@ -85,3 +89,43 @@ windows/config/: windows/src/desktop/**
 windows/engine/: windows/src/engine/**
 # windows/resources/
 # windows/samples/
+
+#
+# The lower part of the file adds labels based on the PR title
```
Comment (mcdurdin) [inline .github/labeler.yml:94]:
```suggestion
# Add labels based on the PR title using jimschubert/labeler-action
```

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Thank you for this PR. This is an itch that I've been considering scratching for a while now.

But I don't think that what this PR has right now is going to be adequate.

I am not comfortable with the fact that jimschubert/labeler-action uses the same config file as GitHub's actions/labeler (.github/labeler.yml), but in a very different and incompatible way. This doesn't feel like a good thing to be doing...

More importantly, though, I don't want labels to be added based on the description text for an issue or PR, but **only** for the metadata found in the issue/PR title. So in order for this to be a goer, I think we need to either look for an alternative action or write our own. Given that jimschubert/labeler-action is written in Go and we have no other Go code, I am not keen on forking and maintaining an action based on that one.

File: .github/labeler.yml (hunk 2 above)
Comment (ermshiperete) [reply .github/labeler.yml:131]: This is useful if a change is e.g. for Linux, but it happens in `common/core`, e.g. packaging related. Or a linux specific change under `resources` or `docs`.

**Review by mcdurdin — APPROVED** [review]
Looks better.

My only concern now is that the multi-labeler action is only a few days old so may have stability/trust issues.

The commit you reference is already out of date, although it looks like subsequent commits may not really impact us at this point.

Let's see how it goes.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.11-alpha](https://keyman.com/downloads/releases/alpha/15.0.11)

**Pull request merged**